### ISSUE TITLE
test_mock_build failures

### ISSUE THREAD

**Issue by sbailey** [desihub/desitarget#568]
tests are failing at NERSC on desitarget master:

```
======================================================================
ERROR: test_targets_truth (desitarget.test.test_mock_build.TestMockBuild)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/global/common/software/desi/cori/desiconda/20190804-1.3.0-spec/code/desitarget/master/py/desitarget/test/test_mock_build.py", line 43, in test_targets_truth
    targets_truth(params, healpixels=[99737,], nside=256, output_dir=self.outdir, no_spectra=True)
  File "/global/common/software/desi/cori/desiconda/20190804-1.3.0-spec/code/desitarget/master/py/desitarget/mock/build.py", line 976, in targets_truth
    qso_selection='colorcuts', mockdata=mockdata)
  File "/global/common/software/desi/cori/desiconda/20190804-1.3.0-spec/code/desitarget/master/py/desitarget/io.py", line 436, in write_targets
    drint = int(indir.split("dr")[1][0])
AttributeError: 'NoneType' object has no attribute 'split'
```

@moustakas could you fix this quickly or otherwise punt it back?

**Comment by geordie666** [comment]
I think the fix is just to add `AttributeError` to the tuple of exceptions here:

https://github.com/desihub/desitarget/blob/57e151e037d452078d3a848eeaf508b27ec4f390/py/desitarget/io.py#L435-L439

Some context: As Gaia-only targets don't have a release number, I needed a way to find the intended data release outside of the Legacy Surveys footprint, or fail gracefully. Looks like I forgot to catch the `NoneType` possibility in the exception.

**Comment by moustakas** [comment]
What @geordie666 said.

**Comment by geordie666** [comment]
I have fixed this in my working branch, which I hope to have a PR for on Friday afternoon. I can fix this earlier, if needed.

**Comment by sbailey** [comment]
NERSC is out today so it isn't critical to get a PR today, but one tomorrow morning would be tangibly more useful than one tomorrow afternoon, if possible.  Once NERSC is back I'd like to proceed with integration testing and get a clean slate of tags for the OSU workshop, and this issue is currently a blocking factor for the integration testing.

**Comment by geordie666** [comment]
This should be fixed in #570 (which also includes a number of other minor bug-fixes).
